worlds/stardew_valley/options.py

```diff
@@ -533,6 +533,19 @@ class GiftTax(SpecialRange):
     }
 
 
+class Mods(OptionSet):
```

Comment:
Could be doable and thankfully Nexus stores past versions in the case of a problem.  

How the mod string has been working is that I've been using exactly the name of the mod as seen on the Nexus, in order for users to better understand what to type in to turn on the desired mod (lack of a list and toggles and all that without bloat etc).  

Likely we could include a constructed list of version numbers and send this data along with the mod name data, perhaps even modify the mod data that gets sent as a dictionary, and does a comparison.  Though how this comparison will be done I'll have to think about since I think what you're wanting is that also on the *client side* it will stop the player from playing if the mod is the wrong version, which will require us to be able to read what version of the mod the system is running.

Reply:
The client side validation, I wouldn't worry about it immediately, I'd work on that before public release. For the beta, it's fine tbh. This was more of a suggestion for a backend side structure for version numbers, that we would be maintaining on our own, and yes, send to the client in the slotData.

I wouldn't ask the user for the mod version, just the mod name and we'd add the version we support automatically

Reply:
I've added a new file, mod_data.py, which includes current versions I'm using for each mod, and in the init file I've made it so it sends this dictionary to the game.